lib/ImageProcessor.js

```diff
@@ -85,13 +88,18 @@ class ImageProcessor {
             resize.bucket = resize.bucket || bucket;
             resize.jpegOptimizer = resize.jpegOptimizer || jpegOptimizer;
 
-            promise = promise.then(() => this.execResizeImage(resize, imageData).then((image) => {
-                this.fileSystem.putObject(image);
-            }));
-            processedImages++;
+            promise = promise
+              .then(() => this.execResizeImage(resize, imageData))
+              .then((image) => this.fileSystem.putObject(image))
+              .then(() => Promise.resolve(++processedImages));
         });
 
-        return promise.then(() => new Promise((resolve) => resolve(processedImages)));
+        if ( removeAfterSuccess ) {
```

Comment:
- I don't see any new unit tests for this part of `ImageProcessor`, that's one.
- Second, I wonder why it's placed here and not inside `if ( config.exists("backup") ) {...}` ?
- Third, I believe that last `Promise` returned from this method should carry number of processed images.